src/lib/utils/routes.ts

```diff
@@ -8,13 +8,39 @@ export const routeMatchesUrl = (url: string, route: NavMenuItem): boolean => {
   return !!url.match(new RegExp(`^${escapeRegExp(routeUrl)}\/?(\\?|#|$)`, 'i'));
 }
 
+/**
+ * Parses the passed nav menu items and
+ * based on the `filter` param
+ * filters out or includes a route
+ * @param navigationItem navigation menu
+ * @param filter Filter function
+ *
+ * @returns NavMenuItem
+ */
+export const filterRoutes = (navMenuItem: NavMenuItem, filter: (n: NavMenuItem) => boolean, depth?: number) => {
+  // safe escape if things get out of control
+  if (depth >= 9) {
+    return
+  }
+
+  if (!filter(navMenuItem)) {
+    return
+  }
+
+  return {
+    ...navMenuItem,
+    children: navMenuItem.children?.map((childNavMenuItem) => (
+      filterRoutes(childNavMenuItem, filter, depth + 1)
+    )).filter(Boolean)
```

Comment:
Is there a case where the parent menu would be shown based on role but the children might not be?

Reply:
Not now, no. But it could be. 

Also, rather than hardcoding and checking only first level, It was just easier and better to handle it recursively so any child/parent can be hidden if needed/wanted.